### PR TITLE
Improved parser and tests

### PR DESCRIPTION
**Pull request by chevtche**

### CODE REVIEW

**Comment by eile** [comment]
@chevtche : Please cherry pick 5659c31 from my fork, review and test it.

**Comment by eile** [comment]
Improved in 7563673

File: brion/target.h [hunk 1]
```diff
@@ -63,6 +63,9 @@ class Target
      */
     BRION_API const Strings& getTargetNames( const TargetType type ) const;
 
+    /** @return true of the target exists. */
```
Comment (tribal-tec) [inline brion/target.h]: if, not of
missing @version 1.7

File: brion/target.cpp [hunk 2]
```diff
@@ -137,49 +147,55 @@ const Strings& Target::getTargetNames( const TargetType type ) const
...
+
+    GIDSet gids;
+    Strings names( 1, root );
+    while( !names.empty( ))
     {
-        try
+        const std::string name = names.back();
+        names.pop_back();
+
+        if( name[0] == 'a' ) // maybe a GID
```
Comment (tribal-tec) [inline brion/target.cpp]: name.empty() cannot happen I hope

File: brion/target.cpp [hunk 3]
```diff
@@ -25,6 +25,7 @@
 #include <boost/lexical_cast.hpp>
 #include <boost/regex.hpp>
 #include <fstream>
+#include <iterator>
```
Comment (tribal-tec) [inline brion/target.cpp]: used where?

File: brion/target.cpp (hunk 2 above)
Comment (chevtche) [inline brion/target.cpp]: Done

File: brion/target.h (hunk 1 above)
Comment (chevtche) [inline brion/target.h]: Done

File: brion/target.cpp (hunk 3 above)
Comment (eile) [inline brion/target.cpp]: https://github.com/BlueBrain/Brion/pull/56/files/30a84df6e7b48ae9e7cfda75e1af5511342c5f9b#diff-cc8e7494b2cda66f0fb6db6c0288a137R190

File: brion/target.h [hunk 4]
```diff
@@ -63,6 +63,9 @@ class Target
      */
     BRION_API const Strings& getTargetNames( const TargetType type ) const;
 
+    /** @return true of the target exists. @version 1.7 */
```
Comment (tribal-tec) [inline brion/target.h]: still if, not of!

**Comment by chevtche** [comment]
Sorry for the "if"... Done

**Pull request merged**